src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -451,7 +451,10 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			let newConnectionProfile = new ConnectionProfile(this.notebookOptions.capabilitiesService, newConnection);
 			this._activeConnection = newConnectionProfile;
 			this.refreshConnections(newConnectionProfile);
-			await this._activeClientSession.updateConnection(this._activeConnection);
+			this._activeClientSession.updateConnection(this._activeConnection.toIConnectionProfile()).catch(() => {
+				let kernelDisplayName = this.getDisplayNameFromSpecName(this.clientSession.kernel.name);
+				this.notifyError(localize('connectionNotValid', 'Connection is not valid for the kernel: {0}. Choose a different one', kernelDisplayName));
```

Comment:
As mentioned in the other review, what we should do is capture the error sent back from the extension (which knows what went wrong) and show to the user.